### ISSUE TITLE
CSIStorageCapacity deprecation not found

### ISSUE THREAD

**Issue by jkleinlercher** [rikatz/kubepug#407]
According to https://kubernetes.io/blog/2023/03/17/upcoming-changes-in-kubernetes-v1-27/#removal-of-storage-k8s-io-v1beta1-from-csistoragecapacity the 

```
apiVersion: storage.k8s.io/v1beta1
kind: CSIStorageCapacity
```

is deprecated since v1.24. However when running kubepug it doesn't recognize it as deprecated.
When running kubepug in debug mode I find the log entry:

`Adding storage.k8s.io/v1beta1/CSIStorageCapacity to map. Deprecated: false`

command: 

`kubepug -v debug --force-download --k8s-version=v1.25.0 --input-file csi-storage-capacity-deprecated.yaml`

Any idea?

**Comment by rikatz** [comment]
hey @jkleinlercher thanks for using the project.

Yes, so the big issue today is that we rely on swagger.json to detect deprecated fields. This usually should be added to the description of the field, but I have just checked and on swagger of v1.24 the CSIStorage is not marked as deprecated.

I'm still thinking of a better way to populate and detect deprecated APIs other than just looking into swagger.json and will take a look again if I can vendor different APIs versions to check it

**Comment by rikatz** [comment]
@jkleinlercher btw your request gave me an idea, so recent APIs are marked with the warnings:

https://github.com/search?q=repo%3Akubernetes%2Fkubernetes%20%2F%2F%20%2Bk8s%3Aprerelease-lifecycle-gen&type=code

I will add a new database input that allows importing a specific version of Kubernetes types and checking it, but this will probably be deferred to v1.6 :)

**Comment by rikatz** [comment]
Closed in #419
